src/main/java/bio/terra/common/tracing/OkHttpTracingExtractor.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.common.tracing;
+
+import io.opencensus.contrib.http.HttpExtractor;
+import javax.annotation.Nullable;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class OkHttpTracingExtractor extends HttpExtractor<Request, Response> {
+
+  @Nullable
+  @Override
+  public String getRoute(Request request) {
+    // OpenCensus spec wants this to be something like the literal string
+    // "/api/workspace/{workspaceId}", whereas path would be "/api/workspace/12345".
+    // We don't have the route available and this is a best-effort method, so use path instead.
+    return request.url().encodedPath();
+  }
+
+  @Nullable
+  @Override
+  public String getUrl(Request request) {
+    return request.url().toString();
+  }
+
+  @Nullable
+  @Override
+  public String getHost(Request request) {
+    return request.url().host();
+  }
+
+  @Nullable
+  @Override
+  public String getMethod(Request request) {
+    return request.method();
+  }
+
+  @Nullable
+  @Override
+  public String getPath(Request request) {
+    return request.url().encodedPath();
+  }
+
+  @Nullable
+  @Override
+  public String getUserAgent(Request request) {
+    return request.header("user-agent");
+  }
+
+  @Override
+  public int getStatusCode(@Nullable Response response) {
+    return response != null ? response.code() : 0;
```

Comment:
Is this supposed to be an HTTP response code? Maybe returning -1 would be less ambiguous, though I guess 0 isn't a valid response code either.

Reply:
This is normally an HTTP status code, though the doc on the method I'm implementing explicitly requires "If the response is null, this method should return 0". Added that as a comment as well since it's a little strange

Reply:
I'd probably invert the ternary so that it's one less backflip for the brain.